File: packages/core/src/sql-normalizer.ts
export function normalizeSQLFragment(value: string): string {
  return value.replace(/\s+/g, ' ').trim()
}

export function normalizeEngine(engine: string): string {
  let normalized = engine.trim().replace(/^Shared/, '')
  if (!normalized.includes('(')) {
    normalized += '()'
  }
  return normalized
}

File: chkit_python/src/chkit/core/sql_normalizer.py
"""Whitespace + engine normalization helpers used during canonicalization."""

from __future__ import annotations

import re
from typing import Final

_WHITESPACE: Final[re.Pattern[str]] = re.compile(r"\s+")


def normalize_sql_fragment(value: str) -> str:
    return _WHITESPACE.sub(" ", value).strip()


def normalize_engine(engine: str) -> str:
    normalized = engine.strip()
    if normalized.startswith("Shared"):
        normalized = normalized[len("Shared") :]
    if "(" not in normalized:
        normalized += "()"
    return normalized
